src/layouts/Sidebar.jsx

```diff
@@ -8,127 +8,214 @@ import { useAuth } from "../context/AuthContext";
+  const [isActive, setIsActive] = useState({
+    home: false,
+    application: false,
+    session: false,
+    module: false,
+    settings: false,
+  });
+
+  const [isSubActive, setIsSubActive] = useState({
+    task: false,
+    assessment: false,
+    quiz: false,
+  });
+  const [isModuleActive, setIsModuleActive] = useState(false);
+
+  const handleMainLinkClick = (link) => {
+    const updateActive = { ...isActive };
+    Object.keys(updateActive).forEach((key) => {
+      const checkIsActive = key === link ? true : false;
+      updateActive[key] = checkIsActive;
+    });
+    setIsActive(updateActive);
+    if (link === "module") {
+      setIsModuleActive(!isModuleActive);
     }
-    axios.get(`${API_END_POINT}/api/list/batch/`,{headers}).then((res)=>{
-      setBatches(res.data.data);
-    })
-  }, []);
+  };
 
+  const handleSubLinkClick = (link) => {
+    const updateSubActive = { ...isSubActive };
+    Object.keys(updateSubActive).forEach((key) => {
+      const checkIsSubActive = key === link ? true : false;
+      updateSubActive[key] = checkIsSubActive;
+    });
+    setIsSubActive(updateSubActive);
+  };
   return (
-    <div className="sidebar">
+    <nav className="side-nav-container flex">
       <div className="logo">
-        <img src={dckapLogo} alt="dckap-logo-sm" />
+        <img src="/images/dckap_palli_logo_sm.svg" alt="DCKAP Palli logo" />
+      </div>
+      <div className="batch-switch-container flex">
+        <div className="batch-content-container flex">
+          <div className="batch-logo">
+            <p>B1</p>
+          </div>
+          <div className="batch-name">
+            <p>Batch 1</p>
+            <span>2023-2024</span>
+          </div>
+        </div>
+        <div className="switch-icon">
+          <img src="/icons/dropdown.svg" alt="" />
+        </div>
       </div>
 
-      <div className="menus">
-        {!isDashboardPage && (
-          <Link to="/dashboard">
-            <Tooltip title="Dashboard" placement="right" color={"#223F64"}>
-              <div className="dashboard-container menu-container">
-                <div className="applicants flex">
-                  <span className="material-symbols-outlined">arrow_back</span>
-                  <p>Back Dashboard</p>
-                </div>
-              </div>
-            </Tooltip>
-          </Link>
-        )}
-
-        {menuList.map((menu, index) => (
-          <Link
-            to={isDashboardPage ? "/dashboard" : `/batch/${batchId}/${menu.id}`}
-            key={index}
-            onClick={() => setActive(menu.id)}
-          >
-            <Tooltip title={menu.label} placement="right" color={"#223F64"}>
-              <div
-                className={`${menu.id}-container menu-container ${
-                  menu.id === active ? "active" : ""
+      <div className="nav-links">
+        <ul>
+          <li className={`main-link ${isActive.home ? "main-active" : ""}`}>
+            <a
```

Comment:
It would better if you better render these in an array
for example
`[{label:"Applications",id:"application},{label:"Module",id:"module,"subMenu":[{label:"Task",id:"task}]}]`